safekeeper/src/wal_backup.rs

```diff
@@ -494,15 +494,13 @@ async fn backup_object(
         .as_ref()
         .unwrap();
 
-    let file = tokio::io::BufReader::new(
-        File::open(&source_file)
-            .await
-            .with_context(|| format!("Failed to open file {} for wal backup", source_file))?,
-    );
-
-    storage
-        .upload_storage_object(Box::new(file), size, target_file)
+    let file = File::open(&source_file)
         .await
+        .with_context(|| format!("Failed to open file {source_file:?} for wal backup"))?;
+
+    let file = tokio_util::io::ReaderStream::with_capacity(file, 8 * 1024);
```

Comment:
Most likely Safekeepers would also benefit from larger buffer size?

Reply:
I was hoping to clean this PR up to have just the current values, then maybe raise them in a separate one.

Reply:
I split at least the 32kB of pageserver side, need to re-check the whole PR, unresolving this on purpose.